src/boot/efi/meson.build

```diff
@@ -102,6 +102,46 @@ if have_gnu_efi
         efi_conf.set10('ENABLE_TPM', get_option('tpm'))
         efi_conf.set('SD_TPM_PCR', get_option('tpm-pcrindex'))
 
+        if get_option('sbat_distro') != ''
+                efi_conf.set_quoted('SBAT_PROJECT', meson.project_name())
+                efi_conf.set_quoted('PROJECT_VERSION', substs.get('PROJECT_VERSION'))
+                efi_conf.set_quoted('PROJECT_URL', substs.get('PROJECT_URL'))
+                efi_conf.set_quoted('SBAT_DISTRO_GENERATION', '@0@'.format(get_option('sbat_distro_generation')))
+                sbatvars = [['sbat_distro', 'ID'],
+                            ['sbat_distro_summary', 'NAME'],
+                            ['sbat_distro_url', 'BUG_REPORT_URL']]
+                foreach sbatvar : sbatvars
+                        value = get_option(sbatvar[0])
+                        if value == '' or value == 'auto'
+                                value = run_command('sh', '-c', '. /etc/os-release && echo $' + sbatvar[1]).stdout().strip()
+                        endif
+                        if value != ''
+                                efi_conf.set_quoted(sbatvar[0].to_upper(), value)
+                        endif
+                endforeach
+
+                sbatvars = [['sbat_distro_pkgname', meson.project_name()],
+                            ['sbat_distro_version', get_option('version-tag')]]
+                foreach sbatvar : sbatvars
+                       value = get_option(sbatvar[0])
+                       if value == ''
+                               value = sbatvar[1]
+                       endif
+                       efi_conf.set_quoted(sbatvar[0].to_upper(), value)
+                endforeach
+                if efi_conf.get('SBAT_DISTRO_VERSION') == '""'
+                       efi_conf.set_quoted('SBAT_DISTRO_VERSION', substs.get('PROJECT_VERSION'))
+                endif
+                required_sbatvars = ['sbat_distro', 'sbat_distro_generation', 'sbat_distro_summary',
+                                     'sbat_distro_pkgname', 'sbat_distro_version', 'sbat_distro_url' ]
+                foreach sbatvar : required_sbatvars
+                       value = efi_conf.get(sbatvar.to_upper())
+                       if value == '' or value == '""' or value == '"auto"'
```

Comment:
How could we get `""`? The values are generated with `echo` above, so if we get `""` here it would mean some strange issue with quoting. Also, where do the quotes around `auto` come from?

Reply:
set_quoted generates `""` value for empty string `"auto"` for `auto` etc.

Reply:
Oh, that's messy.

Why not write it like this then:
```python
               foreach tuple : [
                           # option name, os-release fallback, required
                           ['sbat-distro',                   'ID',        true],
                           ['sbat-distro-summary', 'NAME', true],
                           ['sbat-distro-url', 'BUG_REPORT_URL', false]]
                       option_name, fallback_name, required = tuple
                       value = get_option(option_name)
                       if value == '' or value == 'auto'
                               value = run_command('bash', '-c', ...)
                               if value == '' and required
                                        error(...)
                               endif
                       endif
                       if value != ''
                            efi_conf.set_quoted(sbatvar[0].underscorify().to_upper(), value)
                       endif
             endforeach
```